staging/src/k8s.io/apimachinery/pkg/api/operation/context.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operation
+
+// Context provides contextual information about a request validation.
```

Comment:
nite: not just about validation, but about an API operation.  Iam assuming the intent of moving this out of validation-territory is to make it more plausibly generic?  Is this the type where we will pass the "opts" ?  If so, is it really generic or is it highly coupled to validation?  Sorry to be waffling, just trying to figure where it fits.

What's the support commitment for k8s.io/apimachinery?  Is it as strong as k8s.io/utils or as weak as k/k ?

We can do this for now and re-eval as we push ahead

Reply:
I'd tucked it away in validation initially because I expected we'd want to customize it for validation needs.  I think I'd rather move it back under validate than try to make something more generic.

Backward compatibility it going to be quite strict here.  I've noticed that in the ecosystem, it's quite common to have significant skew between generated code and libraries.  For example, someone might generate a client and the use that generated client while still upgrading their k/k libs regularly.  So long as we retain backward compatibility in the libraries that generated code depends on, this works.  So I think we must operate more like `utils` here in terms of support.

> We can do this for now and re-eval as we push ahead

I'll clean up the godoc and add a comment.